tests/test_testing.py

```diff
@@ -0,0 +1,157 @@
+from typing import Dict, Optional
+
+import pytest
+
+from blacksheep import Content
+from blacksheep.contents import JSONContent
+from blacksheep.server.application import Application
+from blacksheep.server.bindings import FromHeader
+from blacksheep.server.responses import Response
+from blacksheep.testing import AbstractTestSimulator, TestClient
+
+
+class CustomTestSimulator(AbstractTestSimulator):
+    async def send_request(
+        self,
+        method: str,
+        path: str,
+        headers: Optional[Dict[str, str]] = None,
+        query: Optional[Dict[str, str]] = b"",
+        content: Optional[Content] = None,
+    ):
+        if method == "GET":
+            return {"custom": "true"}
+
+
+async def _start_application(app):
+    await app.start()
+
+
+@pytest.fixture
+def test_app():
+    return Application(show_error_details=True)
```

Comment:
Since pytest supports asynchronous fixtures, maybe we could remove the calls to `_start_application` and use instead:

```python
@pytest.fixture
async def test_app():
    app = Application(show_error_details=True)
    await app.start()
    yield app
    await app.stop()  # necessary only if the app needs to close gracefully things when it stops, not in this specific case
```

Reply:
That's a good point, I've tried the same example as you wrote, but tests failing with the next error
```
TypeError: Cannot convert dict to blacksheep.messages.Response
```

An interesting thing that in the tests we receiving a started application, for some reason, if we don't start the application within a test, it fails, thoughts?

Reply:
Absolutely! I made a mistake - we cannot start the application in a fixture because you need to register request handlers as part of the test arrangements _before_ starting the application! Your code is good as it is now and it cannot work otherwise.

Reply:
note: I marked it again as "unresolved" only because the GitHub UI hides too resolved conversations too much and I think they are almost impossible to see. 😄